source/mesh_deformation/ascii_data.cc

```diff
@@ -0,0 +1,153 @@
+/*
+  Copyright (C) 2016 - 2019 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <aspect/global.h>
+#include <aspect/gravity_model/interface.h>
+#include <aspect/mesh_deformation/ascii_data.h>
+#include <aspect/geometry_model/interface.h>
+
+#include <deal.II/base/parameter_handler.h>
+#include <array>
+
+
+namespace aspect
+{
+  namespace MeshDeformation
+  {
+    template <int dim>
+    AsciiData<dim>::AsciiData ()
+      :
+      surface_boundary_id(1)
+    {}
+
+
+
+    template <int dim>
+    void
+    AsciiData<dim>::initialize ()
+    {
+      surface_boundary_id = this->get_geometry_model().translate_symbolic_boundary_name_to_id("top");
+
+      std::set<types::boundary_id> surface_boundary_set;
+      surface_boundary_set.insert(surface_boundary_id);
+
+      Utilities::AsciiDataBoundary<dim>::initialize(surface_boundary_set,
+                                                    1);
+    }
+
+
+
+    template <int dim>
+    Tensor<1,dim>
+    AsciiData<dim>::compute_initial_deformation_on_boundary(const types::boundary_id boundary_indicator,
+                                                            const Point<dim> &position) const
+    {
+      const double topo = Utilities::AsciiDataBoundary<dim>::get_data_component(boundary_indicator,
+                                                                                position,
+                                                                                0);
+
+      const Tensor<1,dim> gravity = this->get_gravity_model().gravity_vector(position);
+
+      Tensor<1,dim> topography_direction;
+      if (gravity.norm() > 0.0)
+        topography_direction = -gravity / gravity.norm();
+
+      return topo * topography_direction;
+    }
+
+
+
+    template <int dim>
+    Tensor<1,dim-1>
+    AsciiData<dim>::vector_gradient(const Point<dim> &point) const
```

Comment:
Why do you need this function? I would think everything works after deleting it.

Reply:
It was copied from the geometry ascii data implementation. I deleted it now.

Reply:
It looks like you only deleted the implementation in the .cc file, but you still have the declaration in the .h file. Please remove that as well. Afterwards this looks good to go.

Reply:
OK, I fixed that now.